src/WebJobs.Script/Workers/SharedMemoryDataTransfer/MemoryMappedFileAccessor.cs

```diff
@@ -0,0 +1,132 @@
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.SharedMemoryDataTransfer
+{
+    /// <summary>
+    /// TODO gochaudh:
+    /// Update docs. Param docs may have changed.
+    /// </summary>
+    public abstract class MemoryMappedFileAccessor : IMemoryMappedFileAccessor
+    {
+        public MemoryMappedFileAccessor(ILoggerFactory loggerFactory)
+        {
+            Logger = loggerFactory.CreateLogger<MemoryMappedFileAccessor>();
+        }
+
+        protected ILogger Logger { get; }
+
+        public abstract bool TryCreate(string mapName, long size, out MemoryMappedFile mmf);
+
+        public abstract bool TryOpen(string mapName, out MemoryMappedFile mmf);
+
+        public abstract void Delete(string mapName, MemoryMappedFile mmf);
+
+        public bool TryCreateOrOpen(string mapName, long size, out MemoryMappedFile mmf)
+        {
+            if (TryOpen(mapName, out mmf))
+            {
+                return true;
+            }
+
+            if (TryCreate(mapName, size, out mmf))
+            {
+                return true;
+            }
+
+            Logger.LogError($"Cannot create or open: {mapName} with size {size}");
+            return false;
+        }
+
+        /// <summary>
+        /// Get the path of the file to store a <see cref="MemoryMappedFile"/>.
+        /// We first try to mount it in memory-mounted directories (e.g. /dev/shm/).
+        /// </summary>
+        /// <param name="mapName">Name of the <see cref="MemoryMappedFile"/>.</param>
+        /// <returns>Path to the <see cref="MemoryMappedFile"/> if found an existing one
+        /// <see cref="null"/> otherwise.</returns>
+        protected string GetPath(string mapName)
+        {
+            // We escape the mapName to make it a valid file name
+            // Python will use urllib.parse.quote_plus(mapName)
+            string escapedMapName = Uri.EscapeDataString(mapName);
+
+            // Check if the file already exists
+            string filePath;
+            foreach (string tempDir in SharedMemoryConstants.TempDirs)
+            {
+                filePath = Path.Combine(tempDir, SharedMemoryConstants.TempDirSuffix, escapedMapName);
+                if (File.Exists(filePath))
+                {
+                    return filePath;
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Create a path which will be used to create a file backing a <see cref="MemoryMappedFile"/>.
+        /// </summary>
+        /// <param name="mapName">Name of the <see cref="MemoryMappedFile"/>.</param>
+        /// <param name="size">Required size in the directory.</param>
+        /// <returns>Created path.</returns>
+        protected string CreatePath(string mapName, long size)
+        {
+            // We escape the mapName to make it a valid file name
+            // Python will use urllib.parse.quote_plus(mapName)
+            string escapedMapName = Uri.EscapeDataString(mapName);
```

Comment:
We should avoid duplicating this logic

Reply:
Sorry, this was an outdated comment. The intention is to always use UUIDs as map names (on both worker and host) so there is no logic duplication. Will revisit this once we finalize the worker changes also.